src/inmanta/loader.py

```diff
@@ -428,6 +434,57 @@ def _loading_top_level_package(self) -> bool:
         return self.path == ""
 
 
+def strip_py(module: list[str]) -> list[str]:
+    """
+    Strip __init__.py or .py file extension from module parts.
+    """
+    if module == []:
+        return []
+    init, last = module[:-1], module[-1]
+    if last == "__init__.py" or last == "__init__.pyc":
+        return init
+    if last.endswith(".py"):
+        return list(chain(init, [last[:-3]]))
+    if last.endswith(".pyc"):
+        return list(chain(init, [last[:-4]]))
+    return module
+
+
+def split(path: str) -> Iterator[str]:
+    """
+    Returns an iterator over path's parts.
+    """
+    if path == "":
+        return iter(())
+    init, last = os.path.split(path)
+    yield from split(init)
+    if last != "":
+        yield last
+
+
+def convert_module_path_to_namespace(path: str) -> str:
```

Comment:
How is this distinct from the method below?

Reply:
removed